### PR TITLE
Budget App/Aposor

### PR DESCRIPTION
**Pull request by RayhanTabase**
In this PR Implemented the following requirements:
- [Presentation](https://www.loom.com/share/cdb3802b5ad34ea490e9e18dd1de60f0)
- [Live](https://shrouded-garden-02833.herokuapp.com/categories)
- Created entity, user, and group models and tables
- Added category pages where users can view and add categories
- Added transaction pages where users can view and add transactions
- Ability to sign in and sign up
- Need to be signed in to view all pages except the splash page
- Added testing

### CODE REVIEW

File: spec/models/entity_group_spec.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe EntityGroup, type: :model do
+  pending "add some examples to (or delete) #{__FILE__}"
+end
```
Comment (Rustamxon7) [inline spec/models/entity_group_spec.rb:5]: - [ ] Kindly remove the tests if you do not need them. To make your code cleaner.

File: spec/helpers/categories_helper_spec.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+# describe CategoriesHelper do
+#   describe "string concat" do
+#     it "concats two strings with spaces" do
+#       expect(helper.concat_strings("this","that")).to eq("this that")
+#     end
+#   end
+# end
+RSpec.describe CategoriesHelper, type: :helper do
+  pending "add some examples to (or delete) #{__FILE__}"
+end
```
Comment (Rustamxon7) [inline spec/helpers/categories_helper_spec.rb:15]: - [ ] Kindly remove the tests if you do not need them. To make your code cleaner.

File: spec/helpers/home_helper_spec.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+# describe HomeHelper do
+#   describe "string concat" do
+#     it "concats two strings with spaces" do
+#       expect(helper.concat_strings("this","that")).to eq("this that")
+#     end
+#   end
+# end
+RSpec.describe HomeHelper, type: :helper do
+  pending "add some examples to (or delete) #{__FILE__}"
+end
```
Comment (Rustamxon7) [inline spec/helpers/home_helper_spec.rb:15]: - [ ] Kindly remove the tests if you do not need them. To make your code cleaner.

File: spec/helpers/transactions_helper_spec.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+# describe TransactionsHelper do
+#   describe "string concat" do
+#     it "concats two strings with spaces" do
+#       expect(helper.concat_strings("this","that")).to eq("this that")
+#     end
+#   end
+# end
+RSpec.describe TransactionsHelper, type: :helper do
+  pending "add some examples to (or delete) #{__FILE__}"
+end
```
Comment (Rustamxon7) [inline spec/helpers/transactions_helper_spec.rb:15]: - [ ] Kindly remove the tests if you do not need them. To make your code cleaner.

File: spec/views/home/index.html.erb_spec.rb [hunk 5]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe 'home/index.html.erb', type: :view do
+  pending "add some examples to (or delete) #{__FILE__}"
+end
```
Comment (Rustamxon7) [inline spec/views/home/index.html.erb_spec.rb:5]: - [ ] Kindly remove the All pending lists if you do not need them.

File: README.md [hunk 6]
```diff
@@ -1,2 +1,77 @@
...
+# Acknowledgement
+
+- Hat tip to anyone whose code was a source of inspiration.
+- A big thanks to [microverseinc](https://github.com/microverseinc).
+- [Gregoire Vella](https://www.behance.net/gregoirevella) on Behance the author of the [original design](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=)
+
+
+## 📝 License
+
+This project is [MIT](./MIT.md) licensed.
```
Comment (Rustamxon7) [inline README.md:77]: - [ ] [OPTIONAL] Please add MIT license to your project.

**Review by rustam-tolipov — CHANGES_REQUESTED** [review]
Hi @RayhanTabase, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
_Your project is almost complete!_
_No Linters errors_


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by keneogu — APPROVED** [review]
Hi @RayhanTabase,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**